src/ledger.js

```diff
@@ -0,0 +1,131 @@
+// LedgerSystem class
+const BigNumber = require('bignumber.js');
+const Data = require("./database");
+const FIFOQueue = require("./queue");
+const sleep = require('./utils').sleep;
+const groupBy = require('./utils').groupBy;
+const OperationTypes = {
+    TRANSFER: "TRANSFER"
+}
+const OperationStatus = {
+    INIT: "INIT",
+    PROCESSING: "PROCESSING",
+    APPLIED: "APPLIED",
+    REJECTED: "REJECTED"
+}
+const Entities = {
+    OPERATIONS: "operations",
+    ACCOUNTS: "accounts",
+    ENTRIES: "entries"
+}
+class LedgerSystem {
+    // singleton class
+    constructor (databaseConfig) {
```

Comment:
pending (not extreme priority but its important for testing) - 

abstraction around database so that ledger.js doesn't know about the concept of databases.

Basically an abstraction like 'ICacheStorage' in CRUX SDK

Reply:
Actually ICacheStorage type abstraction is already there I see. We need to rename this `storageConfig` and infer the storage module from storageConfig